samza-kv/src/main/scala/org/apache/samza/storage/kv/AccessLoggedStore.scala

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.samza.storage.kv
+
+
+import java.util
+import org.apache.samza.config.StorageConfig
+import org.apache.samza.task.MessageCollector
+import org.apache.samza.util.Logging
+import org.apache.samza.system.{OutgoingMessageEnvelope, SystemStream, SystemStreamPartition}
+import org.apache.samza.serializers._
+
+class AccessLoggedStore[K, V](
+    val store: KeyValueStore[K, V],
+    val collector: MessageCollector,
+    val changelogSystemStreamPartition: SystemStreamPartition,
+    val storageConfig: StorageConfig,
+    val storeName: String,
+    val keySerde: Serde[K]) extends KeyValueStore[K, V] with Logging {
+
+  object DBOperations extends Enumeration {
+    type DBOperations = Int
+    val READ = 1
+    val WRITE = 2
+    val DELETE = 3
+    val RANGE = 4
+  }
+
+  val streamName = storageConfig.getAccessLogStream(changelogSystemStreamPartition.getSystemStream.getStream)
+  val systemStream = new SystemStream(changelogSystemStreamPartition.getSystemStream.getSystem, streamName)
+  val partitionId: Int = changelogSystemStreamPartition.getPartition.getPartitionId
+  val serializer = new LongSerde()
+  val samplingRatio = storageConfig.getAccessLogSamplingRatio(storeName)
+  val rng = scala.util.Random
+
+  def get(key: K): V = {
+    val isRange = false
+    measureLatencyAndWriteToStream(DBOperations.READ, isRange, toBytesOrNull(key), null, store.get(key))
+  }
+
+  def getAll(keys: util.List[K]): util.Map[K, V] = {
+    val isRange = true
+    measureLatencyAndWriteToStream(DBOperations.READ, isRange, null, toBytesKey(keys.iterator()), store.getAll(keys))
+  }
+
+  def put(key: K, value: V): Unit = {
+    val isRange = true
```

Comment:
Why is a `put` on a per-key setting `range = true`? Would this not merely be a per-key operation?